span.go

```diff
@@ -52,6 +53,10 @@ type Span interface {
 	// See LogData for semantic details.
 	Log(data LogData)
 
+	// Return a WireInjector for the given WireEncoding, or nil if the Span
+	// does not support such an encoding.
+	WireInjectorForEncoding(enc WireEncoding) WireInjector
```

Comment:
Ultimately I thought this was better on `Span`. First of all, there's the convenience aspect: the user *definitely* has a `Span` on hand at injection time, and only possibly has a `Tracer` on hand, esp given how annoying Go can be regarding `net.Context` and so forth (and my desire not to paint anyone into a corner with singletons).

Second of all, per Gitter chatting with @tschottdorf, it's quite possible that there will be different concrete impls of `Span` for different sampling use-cases in an OT implementation, so putting the Injector access point here allows these different impls to return respectively different injectors as needed.